### DISCUSSION TITLE
Why doesn't it check out expired root certificate before checking CRL?

### DISCUSSION THREAD

**Discussion by SergeyKiwi** [openssl/openssl#22274] [Q&A]
I created simple PKI. There are root certificate, root CRL, user and revoked user certificates. Importantly, root certificate is expired! I do that:

OpenSSL 3.0.9

```
openssl verify -show_chain -crl_check_all -CRLfile root.pem.crl -CAfile root.pem.cer user.pem.cer
C = US, O = Company, CN = Root CA
error 10 at 1 depth lookup: certificate has expired
error user.pem.cer: verification failed

openssl verify -show_chain -crl_check_all -CRLfile root.pem.crl -CAfile root.pem.cer revoked.pem.cer
C = US, O = Company, CN = Revoked
error 23 at 0 depth lookup: certificate revoked
error revoked.pem.cer: verification failed
```

Why doesn't it check out root certificate before checking CRL?

**Comment by paulidale** [comment]
It looks like the CRL is triggering on a certificate before it gets to the root one.

Once that fails, not triggering on the root seems quite reasonable.
